Sources/SCNPath/RKPathEntity.swift

```diff
@@ -0,0 +1,215 @@
+//  Created by Grant Jarvis on 3/13/21.
+//  Copyright © 2021 Max Cobb. All rights reserved.
+//
+
+import SceneKit
+import Combine
+import RealityKit
+
+public class RealityKitPathEntity : SCNPathNode {
+    
+    ///The SceneKit scene that this node belongs to.
+    public weak var scene : SCNScene?
+    
+    ///A realityKit entity which represents the generated path.
+    public var realityKitPath = Entity()
+    
+    ///The number of times we have generated a .usdz file
+    ///This is used for changing the file path each time.
+    private var entityCount = 0
+    
+    private var circleEntity : Entity?
+    
+    private var rectangleEntity : Entity?
+    
+    private var pathEntitiesInScene = [Entity]()
+    
+    override init(
+        path: [SCNVector3], width: Float = 0.5,
+        curvePoints: Float = 8, materials: [SCNMaterial] = []
+    ) {
+        super.init(path: path,
+                   width: width,
+                   curvePoints: curvePoints,
+                   materials: materials)
+        do {
+            let shapesScene = try ShapesProj.loadShapesScene()
+            loadShapes(scene: shapesScene)
+        } catch {
+            print(error.localizedDescription)
+            return
+        }
+        //Remove synchronization to save memory.
+        self.realityKitPath.visit(using: {$0.synchronization = nil})
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    
+    private func loadShapes(scene: ShapesProj.ShapesScene){
+        
+        //Change pivot point.
+        rectangleEntity = Entity()
+        rectangleEntity?.addChild(scene.rectangle!)
+        
+        circleEntity = scene.circle
+        
+        //Then we can use custom materials (even with transparency) and video materials.
+        //We could also smooth the points ourselves. They are all on the 2D X-Z plane.
+    }
+    
+    override func recalcGeometry() {
+        guard realityKitPath.isAnchored,
+              circleEntity != nil,
+              rectangleEntity != nil
+        else {return}
+        
+        //Start off with just a circle for the first position,
+        //Then every position after that gets a rectangle and a circle.
+        //This way, every rectangle has a circle on both ends.
+        if self.path.count == 1 {
+            let circleLocal = circleEntity!.clone(recursive: true)
+            realityKitPath.anchor!.addChild(circleLocal)
+            circleLocal.setPosition(self.path[0].getSIMD3Float(), relativeTo: nil)
+            pathEntitiesInScene.append(circleLocal)
+            return
+        }
+        
+        //Using a smaller distance between points on a curve will lead to smoother curves.
+        
+        //Keep all y-values the same so that the edges of the circles don't stick out.
+        let currentIndex = (path.count - 1)
+        let newPosition : SIMD3<Float> = [path[currentIndex].getSIMD3Float().x,
```

Comment:
we shouldn't need getSIMD3Float here though, path[currentIndex].x should be fine?

Reply:
Since we are using SCNVector3's inside of self.hitPoints and pathNode.path, we are converting back and forth. We might rather make different arrays for the RealityKit implementation and just use SIMD3's exclusively. Please let me know which you prefer.